umi_tools/count_tab.py

```diff
@@ -0,0 +1,131 @@
+'''
+count_tab.py - Count reads per gene from flatfile using UMIs
+=================================================================
+
+:Author: Ian Sudbery, Tom Smith
+:Release: $Id$
+:Date: |today|
+:Tags: Python UMI
+
+Purpose
+-------
+
+The purpose of this command is to count the number of reads per gene
+based on the read's gene assignment and UMI. Note this command is not
+currently able to perform per-cell counting. See the count command if
+you want to perform per-cell counting.
+
+The input must be in the following format (tab separated), where the
+first column is the read identifier and the second column is the
+assigned gene. The input must be sorted by the gene identifier:
+
+NS500668:144:H5FCJBGXY:2:22309:18356:15843_TCTAA    ENSG00000279457.3
+NS500668:144:H5FCJBGXY:3:23405:3971:19716_CGATG     ENSG00000225972.1
+
+You can perform any required file transformation and pipe the output
+directly to count_tab. For example to pipe output from featureCounts
+with the '-R' option you can do the following:
+
+    awk '$2=="Assigned" {print $1"\t"$3}' my.bam.featureCounts| sort -k2 |
+    umi_tools count_tab -S gene_counts.tsv -L count.log
+
+The tab file is assumed to contain each read id once only. For paired
+end reads with featureCounts you must include the "-p" option so each
+read id is included once only.
+
+'''
+
+import sys
+import collections
+import re
+
+# required to make iteritems python2 and python3 compatible
+from builtins import dict
+
+from functools import partial
+
+import pysam
```

Comment:
Looks as if quite a lot of these imports are not neccesary. I don't see pysam used anywhere. Nor pandas, nor numpy.